File: java/calculadora.java
import java.util.Scanner;

class SistemaNotas {

    public static float mediaNotas(int[] notas) {
        float soma = 0;
        for (int n : notas) {
            soma += n;
        }
        return soma / 4f;
    }

    public static void verificaAprovacao(int[][] notas, float notaMedia) {
        for (int i = 0; i < notas.length; i++) {
            float media = mediaNotas(notas[i]);
            System.out.printf("Aluno %d - Média: %.2f - ", i + 1, media);
            
            if (media >= notaMedia)
                System.out.println("APROVADO ✅");
            else
                System.out.println("REPROVADO ❌");
        }
    }

    public static void printaNotas(int[][] notas) {
        for (int i = 0; i < notas.length; i++) {
            System.out.print("Aluno " + (i + 1) + " - Notas: ");
            for (int j = 0; j < 4; j++) {
                System.out.print(notas[i][j] + " ");
            }
            System.out.println();
        }
    }

    public static int[][] pegaNotas(int qtdAlunos, Scanner sc) {
        int[][] notas = new int[qtdAlunos][4];

        for (int i = 0; i < qtdAlunos; i++) {
            System.out.println("Digite as 4 notas do aluno " + (i + 1) + ":");

            for (int j = 0; j < 4; j++) {
                while (true) {
                    System.out.print("Nota " + (j + 1) + ": ");
                    int valor = sc.nextInt();

                    if (valor >= 0 && valor <= 10) {
                        notas[i][j] = valor;
                        break;
                    }

                    System.out.println("A nota deve estar entre 0 e 10.");
                }
            }
        }
        return notas;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int qtdAlunos;
        while (true) {
            System.out.print("Digite a quantidade de alunos: ");
            qtdAlunos = sc.nextInt();

            if (qtdAlunos > 0) break;
            System.out.println("Quantidade deve ser positiva.");
        }

        int[][] notas = pegaNotas(qtdAlunos, sc);

        while (true) {
            System.out.println("\nMENU");
            System.out.println("1: Mostrar alunos e notas.");
            System.out.println("2: Mostrar aprovados e reprovados.");
            System.out.println("3: Sair");
            System.out.print("Opção: ");

            int opcao = sc.nextInt();

            switch (opcao) {
                case 1:
                    printaNotas(notas);
                    break;

                case 2:
                    float media;
                    while (true) {
                        System.out.print("Digite a média necessária para aprovação: ");
                        media = sc.nextFloat();
                        if (media >= 0 && media <= 10) break;
                        System.out.println("Valor inválido.");
                    }
                    verificaAprovacao(notas, media);
                    break;

                case 3:
                    System.out.println("Saindo...");
                    sc.close();
                    return;

                default:
                    System.out.println("Opção inválida!");
            }
        }
    }
}

File: python/calculadora.py
def media_notas(notas):
    return sum(notas) / 4

def verifica_aprovacao(notas, nota_media):
    for i, notas_aluno in enumerate(notas, start=1):
        media = media_notas(notas_aluno)
        print(f"Aluno {i} - Média: {media:.2f} - ", end="")
        if media >= nota_media:
            print("APROVADO ✅")
        else:
            print("REPROVADO ❌")

def printa_notas(notas):
    for i, notas_aluno in enumerate(notas, start=1):
        print(f"Aluno {i} - Notas:", notas_aluno)

def pega_notas(qtd_alunos):
    notas = []
    for i in range(qtd_alunos):
        print(f"\nDigite as 4 notas do aluno {i + 1}:")
        notas_aluno = []
        for j in range(4):
            while True:
                valor = int(input(f"Nota {j + 1}: "))
                if 0 <= valor <= 10:
                    break
                print("A nota deve ser entre 0 e 10.")
            notas_aluno.append(valor)
        notas.append(notas_aluno)
    return notas

def main():
    while True:
        qtd_alunos = int(input("Digite a quantidade de alunos: "))
        if qtd_alunos > 0:
            break
        print("Quantidade deve ser positiva e não nula.")

    notas = pega_notas(qtd_alunos)

    while True:
        print("\nMENU")
        print("1: Mostrar alunos e notas.")
        print("2: Mostrar aprovados e reprovados.")
        print("3: Sair")

        opcao = int(input("Digite a opção: "))

        if opcao == 1:
            printa_notas(notas)
        elif opcao == 2:
            while True:
                nota_media = float(input("Digite a média necessária para aprovação: "))
                if 0 <= nota_media <= 10:
                    break
                print("Digite um valor válido.")
            verifica_aprovacao(notas, nota_media)
        elif opcao == 3:
            print("Saindo...")
            break
        else:
            print("Opção inválida.")

main()
